src/renderer/components/+user-management/+cluster-role-bindings/dialog.tsx

```diff
@@ -223,16 +237,31 @@ export class ClusterRoleBindingDialog extends React.Component<ClusterRoleBinding
           isMulti
           themeName="light"
           placeholder="Select service accounts ..."
-          options={serviceAccountStore.items.slice()}
-          formatOptionLabel={value => (
+          options={this.serviceAccountOptions}
+          formatOptionLabel={option => (
             <>
               <Icon small material="account_box" />
-              {` ${value.getName()} (${value.getNs()})`}
+              {` ${option.label}`}
             </>
           )}
-          getOptionLabel={value => `${value.getName()} (${value.getNs()})`}
-          onChange={selected => {
-            this.selectedAccounts.replace(selected);
+          onChange={(selected, meta) => {
+            switch (meta.action) {
+              case "clear":
+                this.selectedAccounts.clear();
+                break;
+              case "deselect-option":
+              case "remove-value":
+              case "pop-value":
+                if (meta.option) {
+                  this.selectedAccounts.delete(meta.option.value);
+                }
+                break;
+              case "select-option":
+                if (meta.option) {
+                  this.selectedAccounts.add(meta.option.value);
+                }
+                break;
+            }
```

Comment:
Notice: Open Closed Principle violation.

Reply:
Using a component is an OCP violation?!?!?